1303-x86-boot-txt_early-add-early-TXT-tests-and-restore-M.patch

```diff
@@ -0,0 +1,268 @@
+From 67dc7a18a02417de737314db61fa90f0498beb61 Mon Sep 17 00:00:00 2001
+From: Krystian Hebel <krystian.hebel@3mdeb.com>
+Date: Mon, 17 Apr 2023 20:09:54 +0200
+Subject: [PATCH 03/11] x86/boot/txt_early: add early TXT tests and restore MBI
+ pointer
+
+These tests validate that important parts of memory are protected
+against DMA attacks, including Xen and MBI. Modules can be tested later,
+when it is possible to report issues to user before invoking TXT reset.
+
+TPM event log validation is temporarily disabled due to issue with its
```

Comment:
This will have to be resolved on GRUB's side. It will probably happen after AMD work is done, or at least together with it. (There are 2 corner cases that invalidate easy solutions mentioned in the comment, one each. One is caused by SMM and/or ACPI, the other would break secure launch in iPXE. I don't think we want to support iPXE in the long run, but it is nice for development and testing, at least for AMD).